### DISCUSSION TITLE
Why does useFetch().execute() not throw an error inside try-catch?

### DISCUSSION THREAD

**Discussion by katerlouis** [nuxt/nuxt#26307] [Questions] [answered]
```html
<template>
  <form @submit.prevent="onSubmit">...</form>
</template>
```

```js
const { execute, error, clear } = useFetch('/return-dumm-error-401', {
  immediate: false,
  watch: false,

  method: 'POST',
  body: {
    name: 'Peter',
    age: 78,
  },
});

onBeforeMount(clear);

async function onSubmit() {
  try {
    await execute();

    // if execute() worked, do something amazing!
    // i.e. animating Peter Griffin dance; 
  } 
  catch (error) {
    // this console.error() does not get executed;
    console.error('Something went wrong with Peter...' error);
  }
}
```

Either I don't understand try-catch well enough, or I found a bug / unused potential (?) in `useFetch().execute()`?

Even if the response of useFetch().execute() (inside onSubmit()) gets a 401 response, the Peter Griffin dance still gets animated and the console.error does not fire.

Why is that? Something else does trigger a console.error, just not with my "Peter" text, which makes me believe that my catch() is never reached. 

I assumed `execute()` would throw an error if something goes wrong.

I also tried looking through the source code but coulnd't make sense of it. The architecture and elaborate typescript threw me off (no pun intended :D)

**Comment by TheAlexLichter** [comment, marked as answer]
Execute "just" updates data and error. You should check these values (e.g. in a watch).

**Reply by katerlouis** [reply]
You mean watching the error ref? How would that stop the `onSubmit` code from executing the onSuccses-Peter-Dance?
Could you elaborate?

All I can come up with is this:

```ts
const { execute, error, clear } = useFetch('/return-dumm-error-401', {
  immediate: false,
  watch: false,

  method: 'POST',
  body: {
    name: 'Peter',
    age: 78,
  },
});

async function onSubmit() {
  try {
    await execute();

    // if execute produced an error, `error.value` should hold be updated by now?
    // whats the proper way to check for an error here?
    if (!_.isNull(error.value)) {
      console.log('this error will be logged!');
      return;
      // or?
      // throw new Error(error.value);
    }

    // if execute() worked, do something amazing!
    // i.e. animating Peter Griffin dance; 
  }
}
catch (error) {
  console.error('Peter lost his dancing shoes', error);
}
```

**Reply by TheAlexLichter** [reply]
I would always advocate for using `$fetch` over `useFetch` in this case as you are sending information. This way (also shown [in my video](https://www.youtube.com/watch?v=njsGVmcWviY)) might exist but error handling is trickier.

> You mean watching the error ref?

Yes

>  How would that stop the onSubmit code from executing the onSuccses-Peter-Dance?
Could you elaborate?

It would not. But checking `if(!error.value)` seems reasonable if you really want to stick with useFetch.

**Reply by katerlouis** [reply]
What makes error handling with $fetch easier? You would have to make some kind of `errorMessage`-ref yourself and then set a value to it on error etc.– all of that is done by useFetch out of the box. 

Or what exactly do you mean by "trickier"?
